src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/repo/Repo.java

```diff
@@ -769,15 +773,24 @@ public void reset(String tag)
           throw new RepoTagNotFoundException(
               "Tag " + tag + " not found in repository " + repoInfo.getRepoId());
         }
-        Ref peeled = r.peel(tagRef);
-        ObjectId tagObjectId = peeled != null ? peeled.getPeeledObjectId() : tagRef.getObjectId();
-        ObjectId headObjectId = r.findRef("HEAD").getObjectId();
 
-        RevertCommand rev = git.revert();
-        for (RevCommit c : git.log().addRange(tagObjectId, headObjectId).call()) {
-          rev = rev.include(c);
-        }
-        rev.call();
+        ResetCommand reset = git.reset()
+            .setMode(ResetType.HARD)
+            .setRef(tagRef.getName());
+
+        reset.call();
```

Comment:
I think you need to take a full exclusion lock now - since you don't want anyone to observe the repo in between these calls?

Reply:
Good point, done.